test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -0,0 +1,262 @@
+        cpuset_mom = single_mom.is_cpuset_mom()
+        if cpuset_mom is False:
+            self.create_multi_vnodes(1)
+        self.create_mom_hook()
+
+        j1 = Job(TEST_USER)
+        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
+                           id=single_mom.shortname, max_attempts=3,
+                           interval=2)
+        j1.set_sleep_time(1000)
+        jid = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid)
+
+        self.server.expect(
+            NODE, {ATTR_NODE_state: 'offline'},
+            id=single_mom.shortname, max_attempts=20, interval=2)
+
+        vname = "vnode[0]"
+        if self.is_cray is True:
+            vlist = []
+            first_vnode = 1
+            vnl = self.server.filter(
+                VNODE, {'resources_available.vntype': 'cray_compute'})
+            vlist = vnl["resources_available.vntype=cray_compute"]
+            # Loop through each compute vnode in the list and check if state =
+            # offline
+            for v1 in vlist:
+                if first_vnode == 1:
+                    first_vnode += 1
+                    vname = v1
+                # Check that the vnode is in offline state
+                self.server.expect(
+                    VNODE, {'state': 'offline'}, id=v1, max_attempts=3,
+                    interval=2)
+        elif cpuset_mom is True:
+            first_vnode = 1
+            vnl = self.server.status(NODE)
+            for v1 in vnl:
+                if single_mom.shortname != v1['resources_available.vnode']:
+                    if first_vnode == 1:
+                        first_vnode += 1
+                        vname = v1['resources_available.vnode']
+                    # Check that the vnode is in offline state
+                    self.server.expect(
+                        VNODE, {'state': 'offline'},
+                        id=v1['resources_available.vnode'], max_attempts=3,
+                        interval=2)
+        else:
+            self.server.expect(
+                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[0]',
+                max_attempts=3, interval=2)
+            self.server.expect(
+                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[1]',
+                max_attempts=3, interval=2)
+
+        mom_host = single_mom.shortname
+        pbs_exec = self.server.pbs_conf['PBS_EXEC']
+        pbsnodes_cmd = os.path.join(pbs_exec, 'bin', 'pbsnodes')
+        pbsnodes_reset = pbsnodes_cmd + ' -r ' + mom_host
+
+        # Set mom sync hook timeout to be a low value because if mom fails to
+        # get the hook after disabling it then next sync will happen after
+        # 2 minutes by default and we don't want to wait that long.
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'sync_mom_hookfiles_timeout': '5'})
+        self.server.manager(MGR_CMD_SET, HOOK, {'enabled': 'False'}, id="h1")

```

Comment:
Can you please change "create_mom_hook()" to return the id of the hook that it created, store that in a variable when calling the function and use that variable here instead of "h1"?

Reply:
Since both these tests need to create only one hook I just made it a function for simplicity. If you feel strongly about changing it then I can do that otherwise I'd like to keep it as is.